test_requests.py

```diff
@@ -411,6 +416,21 @@ def test_DIGESTAUTH_QUOTES_QOP_VALUE(self):
         r = requests.get(url, auth=auth)
         assert '"auth"' in r.request.headers['Authorization']
 
+    def test_DIGESTAUTH_THREADED(self):
```

Comment:
I understand where @tardyp is comming from: all of the digest auth related tests are named this way... Would you recommend the name change given that?

Regarding the test reliability, both towards failure or success: this looks like an inherently complex matter. @tardyp any suggestion?